app/javascript/admin/controllers/svg_icon_upload_controller.js

```diff
@@ -0,0 +1,62 @@
+import { Controller } from 'stimulus';
+
+export default class SvgIconUploadController extends Controller {
+  selectSvgIcon(event) {
+    const currentNavigationLinkModal = this.targets.scope.data.element.querySelector(
+      'div',
+    );
+
+    this.clearInvalidIconTypeMessage(currentNavigationLinkModal);
+
+    const svgIconContent = currentNavigationLinkModal.querySelector(
+      '#svg-icon-content',
+    );
```

Comment:
this `querySelector` is not needed if you use Stimulus targets, see above

Reply:
Okay! Thanks a lot for the tip